client-go/controller/client/http.go

```diff
@@ -0,0 +1,138 @@
+package client
+
+import (
+	"bytes"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/deis/deis/version"
+)
+
+// CreateHTTPClient creates a HTTP Client with proper SSL options
+func CreateHTTPClient(sslVerify bool) *http.Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: !sslVerify},
+	}
+	return &http.Client{Transport: tr}
+}
+
+func rawRequest(client *http.Client, method string, url string, body io.Reader, headers http.Header,
+	expectedStatusCode int) (*http.Response, error) {
+
+	req, err := http.NewRequest(method, url, body)
+	req.Header = headers
+
+	res, err := client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != expectedStatusCode {
+		defer res.Body.Close()
```

Comment:
@Joshua-Anderson is right to do it this way. The body only gets closed on the error condition. In all other cases, it's the responsibility of the caller to close the body.